core/templates/domain/exploration/InteractionObjectFactory.ts

```diff
@@ -143,29 +149,32 @@ export class Interaction {
     const traverseSchemaAndConvertSubtitledToDicts = (
         value: Object[] | Object
     ): Object[] | Object => {
-      let result: Object[] | Object;
+      let result!: Object[] | Object;
 
       if (value instanceof SubtitledUnicode || value instanceof SubtitledHtml) {
         result = value.toBackendDict();
       } else if (value instanceof Array) {
         result = value.map(
           element => traverseSchemaAndConvertSubtitledToDicts(element));
       } else if (value instanceof Object) {
+        type KeyOfValue = keyof typeof value;
+        let _result: Record<KeyOfValue, Object> = {};
         result = {};
-        Object.keys(value).forEach(key => {
-          result[key] = traverseSchemaAndConvertSubtitledToDicts(value[key]);
+        let keys = <KeyOfValue[]>Object.keys(value);
+        keys.forEach(key => {
+          _result[key] = traverseSchemaAndConvertSubtitledToDicts(value[key]);
         });
+        result = _result as Object;
       }
 
       return result || value;
     };
 
-    const customizationArgsBackendDict:
-      InteractionCustomizationArgsBackendDict = {};
-    Object.keys(customizationArgs).forEach(caName => {
+    const customizationArgsBackendDict: Record<string, Object> = {};
+    Object.entries(customizationArgs).forEach(([caName, caValue]) => {
       customizationArgsBackendDict[caName] = {
         value: traverseSchemaAndConvertSubtitledToDicts(
-          customizationArgs[caName].value)
+          caValue.value)
```

Comment:
This can be `value: traverseSchemaAndConvertSubtitledToDicts(caValue.value)`

Reply:
done